content/en/docs/Core concepts/metrics.md

```diff
@@ -0,0 +1,125 @@
+
+More specific metrics are also exported on a per-component basis. To that effect, 
+`baker.Input`, `baker.Filter`, `baker.Output` and `baker.Upload` all have a `Stats` 
+method. `Stats` is called every second and the component is expected to return both
+a predefined set of metrics and a [`baker.MetricsBag`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsBag)
+containing metrics of arbitrary name and types.
+
+Let's illustrate this with metrics exported by a filter via 
+[`baker.FilterStats`](https://pkg.go.dev/github.com/AdRoll/baker#FilterStats):
+
+```go
+type FilterStats struct {
+	NumProcessedLines int64
+	NumFilteredLines  int64
+	Metrics           MetricsBag
+}
+```
+
+In this case `NumProcessedLines` must represent the **total** number of processed 
+lines since Baker started, and `NumFilteredLines` is the number of discarded 
+(or filtered) records. Due to hstorical reasons these fields have lines in them
+but they do mean the number of records.
+
+An important point is that `Stats` can be called from any goroutine so it must be
+safe for concurrent use by multiple goroutines. 
+
+```go
+func (i *myFilter) Stats() baker.FilterStats {
+    bag := make(baker.MetricsBag)
+    bag.AddGauge("current_speed", float64(atomic.LoadInt64(&myFilter.speed)))
+
+    return baker.FilterStats{
+        NumProcessedLines: atomic.LoadInt64(&myFilter.totalLines),
+        NumFilteredLines: atomic.LoadInt64(&myFilter.filteredLines),
+        // Metrics could be let to its default value, nil, if not needed.
+        Metrics: bag,
+    }
+}
+```
+
+#### Configuring metrics in TOML
+
+Baker configuration TOML files may have a `[metrics]` section dedicated to the 
+configuration of a metrics client.
+
+`[metrics.name]` specifies the metrics client to use, from the list of all registered [`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient).
+`[metrics.config]` specifies some configuration settings which are specific to the client you're using.
+
+For example, this is what the `[metrics]` section would look like with the *Datadog* metrics client:
+
+```toml
+[metrics]
+name="datadog"
+
+    [metrics.config]
+    host="localhost:8125"                  # address of the dogstatsd client to which send metrics to
+    prefix="myapp.baker."                  # prefix for all exported metric names
+    send_logs=true                         # whether we should log messages (as Dogstatd events) or not 
+    tags=["env:prod", "region:eu-west-1"]  # extra tags to associate to all exported metrics 
+```
+
+#### Implementing a new metrics client
+
+The [metrics example](https://github.com/AdRoll/baker/tree/main/examples/metrics)
+shows an example implementation of [`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient)
+and how to register it within Baker so that it can be selected in the [metrics.name] TOML section.
+
+In order to be selected from TOML, you must prealably register a 
```

Comment:
is `prealably` a common word? I never saw it and also in italian is an almost-unknown word